[GS] New: Python Hello World (#55)
bitsandvolts/bitsandvolts.github.io: _posts/2025-05-12-gs_00010_python-hello-world.md

Now that you’ve successfully written your first Python program, let’s build
something a little more useful: a counter. This simple project will introduce us
to more key concepts in Python, such as variables, loops, functions, and
conditional statements.

### Starting with a simple counter

We can begin by writing a basic counter that prints numbers. Create a new file
_counter.py_ and add the following code:

```python
count = 1
print("Count is:", count)
```

In this case, we define a **variable** called count and set it to 1. The print()
function is used to output the value of count to the screen. Unlike printing a
fixed message like "Hello World", notice that we don't put the variable name in
quotes. Doing so would print the word "count" instead of its value.

> Count is: 1

### Adding a while Loop

Next, let's use a **while loop** to make the counter print numbers from 1 to 5.
A loop helps us automate repetitive tasks, so instead of manually printing each
count, we can write a single block of code that runs until a condition is met.

Here’s how we modify the counter to use a loop:

```python
count = 1
while count <= 5:
    print("Count is:", count)
    count += 1
```

> In this version, the variable count starts at 1. The while loop checks whether

Reviewer: Could we format "count" in some way to make it easier to distinguish from the rest of the sentence? E.g. italics, or inline code, I leave it up to you. We should then maybe apply this to every occurrence (and other code words/ variables).

Author: done

which lines of code belong to the loop. Any indented lines after the while
statement are considered part of the loop and will be repeated each time it
runs. Without the correct indentation, Python won’t know what to include in the
loop and will raise an error.

> Count is: 1<br> Count is: 2<br> Count is: 3<br> Count is: 4<br> Count is: 5

### Encapsulating the Code in a Function

Another useful concept in programming is the **function**. Functions allow you
to group code to be reused in different parts of your program. Let’s put our
counter logic into a function:

```python
def simple_counter(target):
    count = 1
    while count <= target:
        print("Count is:", count)
        count += 1

simple_counter(5)
```

Functions are **defined** by the keyword _def_ followed by a function name.
Here, we additionally add a **parameter** with the name _target_ to set the
upper limit for our counter. Observe that the previous limit _5_ is replaced by
the parameter. Also note that again indentation defines which commands belong to
the function.

At the end, you just have to **call** the function passing a value for the

> parameter. If you pass _5_, the output is exactly the same as with the previous

Reviewer: Maybe give the user an idea to try out the function with a different value and observe the result.

Author: done

   function again
3. **Abstraction**: Functions can be very complex, but often it's not required
   to know how they work. It is sufficient to know what they do and how you can
   use them. Take the print() function as an example. We didn't explain how it
   works, but you still can use it.

### Conditional statements

Next, we want to improve the text output using the following logic:

- If the count has reached the target, print "finally done".
- If the count is greater than or equal to the target minus 2, print "almost
  there".
- Otherwise, print "stay patient".

We can use **if-else** statements to directly translate this logic into
programming code:

```python
def simple_counter(target):
    count = 1
    while count <= target:
        if count == target:
            print("Finally done! Count is:", count)
        elif count >= target - 2:
            print("Almost there! Count is:", count)
        else:
            print("Stay patient! Count is:", count)
        count += 1

simple_counter(5)
```

**if** is used to check for a condition. Note that if you want to check for
equality, you have to use "==" and **not** "=". This is a common beginner

> mistake. **elif** (short for 'else if') is used to check a second condition if

Reviewer: Explaining the difference between assignment and equality operator briefly could maybe help?

Author: I was thinking about this but decided not to do it because it also adds complexity. I added a short sentence.

> mistake. **elif** (short for 'else if') is used to check a second condition if

Reviewer: ```suggestion
mistake. **elif** (short for 'else if') is used to check a second condition in case
```

Author: done

the if condition is false. You can include as many elif clauses as needed.
**else** handles all remaining cases and acts as the "otherwise" in your
program. Again, indentation shows which command belongs to which structure.

> Stay patient! Count is: 1<br> Stay patient! Count is: 2<br> Almost there!
> Count is: 3<br> Almost there! Count is: 4<br> Finally done! Count is: 5

### Comments

> "What the fu\*\* have I done here?" - This is a question you will probably ask

Reviewer: Not sure, but should we be less explicit?
```suggestion
"What the \*\*\*\* have I done here?" - This is a question you will probably ask
```
If you think this is ok, leave it as is. Just a minor comment from my side.

Author: done

> yourself very soon. Sometimes, code is complex and to be able to understand it a

Reviewer: ```suggestion
yourself very soon. Sometimes, code is complex and to be able to understand it in a
```

Author: done

few days (or years) later it is crucial to leave some comments. For functions,
it is good practise to explain what they do in the first line. For other
locations it depends. A general rule is to comment as little as possible but as
much as necessary. There are three major ways of commenting in Python:

```python
def simple_counter(target):
    """
    Prints numbers from 1 up to the given target (inclusive).

>     This is a Docstring, which is often used for documenting functions

Reviewer: I think we should mention somewhere that this is a multi-line comment, in general?

Author: done
